core/storage/suggestion/gae_models.py

```diff
@@ -1990,3 +2000,1275 @@ def export_data(
+            language_code, contributor_id)
+        return cls.get_by_id(entity_id)
+
+    @classmethod
+    def fetch_page(
+        cls,
+        page_size: int,
+        offset: int,
+        sort_by: Optional[str],
+        topic_ids: Optional[List[str]],
+        last_activity: Optional[int],
+        language_code: str
+    ) -> Tuple[Sequence[TranslationSubmitterTotalContributionStatsModel],
+                int,
+                bool]:
+        """Returns the models according to values specified.
+
+        Args:
+            page_size: int. Number of models to fetch.
+            offset: int. Number of results to skip from the beginning of all
+                results matching the query.
+            sort_by: str|None. A string indicating how to sort the result.
+            topic_ids: List[str]|None.
+                List of topics user has contributions.
+            last_activity: int. Number of days within which a user has
+                contributed.
+            language_code: str. Language Code for filter.
+
+        Returns:
+            3-tuple(sorted_results, next_offset, more). where:
+                sorted_results:
+                    list(TranslationSubmitterTotalContributionStatsModel).
+                    The list of models with supplied filters and sort_by.
+                next_offset: int. Number of results to skip in next batch.
+                more: bool. If True, there are (probably) more results after
+                    this batch. If False, there are no further results
+                    after this batch.
+        """
+
+        sort_options_dict = {
+            constants.CD_ADMIN_STATS_SORT_OPTIONS['IncreasingLastActivity']:
+                -cls.last_contribution_date,
+            constants.CD_ADMIN_STATS_SORT_OPTIONS['DecreasingLastActivity']:
+                cls.last_contribution_date,
+            constants.CD_ADMIN_STATS_SORT_OPTIONS['IncreasingPerformance']:
+                cls.recent_performance,
+            constants.CD_ADMIN_STATS_SORT_OPTIONS['DecreasingAccuracy']:
+                -cls.overall_accuracy,
+            constants.CD_ADMIN_STATS_SORT_OPTIONS['IncreasingAccuracy']:
+                cls.overall_accuracy,
+            constants.CD_ADMIN_STATS_SORT_OPTIONS['DecreasingSubmissions']:
+                -cls.submitted_translations_count,
+            constants.CD_ADMIN_STATS_SORT_OPTIONS['IncreasingSubmissions']:
+                cls.submitted_translations_count
+        }
+
+        sort = -cls.recent_performance
+        if sort_by is not None:
+            sort = sort_options_dict[sort_by]
+
+        # The first sort property must be the same as the property to which
+        # an inequality filter is applied. Thus, the inequality filter on
+        # last_activity can not be used here. Learn more about this here
+        # https://cloud.google.com/appengine/docs/legacy/standard/go111/datastore/query-restrictions#properties_used_in_inequality_filters_must_be_sorted_first.
```

Comment:
Add a conclusion at the end: This is why we are handling the last_activity inequality filter separately below.

Reply:
I added the reasoning and conclusion before reference link. PTAL

Reply:
"below" is ambiguous since it can refer to what's just "below" this comment. This is why my original suggestion included "separately".

Reply:
Changed. PTAL.